rclcpp/src/rclcpp/node_options.cpp

```diff
@@ -227,6 +229,19 @@ NodeOptions::enable_topic_statistics() const
   return this->enable_topic_statistics_;
 }
 
+bool
+NodeOptions::enable_service_introspection() const
+{
+  return this->enable_service_introspection_;
+}
+
+NodeOptions &
+NodeOptions::enable_service_introspection(bool enable_service_introspection) {
+  this->node_options_.reset();  // reset node options to make it be recreated on next access.
```

Comment:
This is not necessary if we remove the option from the rcl struct (which I think we can do).

Reply:
I think that it makes more sense to keep the option in the rcl `rcl_node_options_t` struct as it keeps integration more consistent across client libraries

Reply:
Actually, looking at it again I am mistaken. We can remove it from the rcl struct. This would mean that `rcl` supports enabling/disabling service introspection for the lifetime of the node on a per-client/service basis and (as it is currently) `rclpy` and `rclcpp` would make it such that enabling/disabling would work on  _all_ service/clients related to a node instead -- which is what is defined in the REP.